app/views/contexts/summary/list_collector_block.py

```diff
@@ -235,3 +242,22 @@ def get_blocks_for_related_answers(
             blocks.append(mutable_block)
 
         return blocks
+
+    def _get_blocks_for_repeating_blocks(self, list_model: ListModel) -> list[dict]:
+        blocks: defaultdict[ImmutableDict, list[str]] = defaultdict(list)
+        for list_id in list_model:
+            list_answers = [
+                (answer_id, answer_list_id)
+                for (answer_id, answer_list_id) in self._answer_store.answer_map
+                if answer_list_id == list_id
+                and self._schema.is_answer_in_repeating_blocks(answer_id)
+            ]
+            for answer_id, answer_list_id in list_answers:
+                block = self._schema.get_block_for_answer_id(answer_id)
+                if block:
+                    blocks[block].append(answer_id)
```

Comment:
We shouldn't do this like this. If 2 list collector have the same list name, this would break. It seems like we are doing all this, just to get a list of blocks we need. It should be all blocks inside `repeating_blocks` for this list name. Any blocks without an answer should display as `No answers provided`.

We care about all blocks, not just the ones that have at least 1 answer.

Therefore, this method can be removed and you can have something like:

Pass in `repeating_blocks` which is in scope.

```PY
    def _get_related_answers(
        self, list_model: ListModel, repeating_blocks
    ) -> dict[str, list[dict]] | None:
        section_id = self._section["id"]

        related_answers = self._schema.get_related_answers_for_list_for_section(
            section_id=section_id, list_name=list_model.name
        )

        blocks: list[dict] = []

        if related_answers:
            blocks += self._get_blocks_for_related_answers(related_answers)

        if repeating_blocks:
            blocks += repeating_blocks
```

Reply:
Great catch, not sure where Iw as going with this looking at it now, change made 👍